blaze/expr/tests/test_strings.py

```diff
@@ -40,13 +42,43 @@ def test_like(ds):
 @pytest.mark.parametrize('ds', dshapes)
 def test_str_upper_schema(ds):
     t = symbol('t', ds)
-    expr_upper = getattr(t, 'name', t).str_upper()
-    expr_lower = getattr(t, 'name', t).str_upper()
+    expr_upper = getattr(t, 'name', t).str.upper()
+    expr_lower = getattr(t, 'name', t).str.lower()
     assert (expr_upper.schema.measure ==
             expr_lower.schema.measure ==
             dshape('%sstring' % ('?' if '?' in ds else '')).measure)
 
 
+def test_str_namespace():
+    t = symbol('t', 'var * {name: string}')
+    assert bzs.str_upper(t.name).isidentical(t.name.str.upper())
+    assert bzs.str_lower(t.name).isidentical(t.name.str.lower())
+    assert (bzs.str_lower(bzs.str_upper(t.name))
+            .isidentical(t.name.str.upper().str.lower()))
+    assert bzs.str_len(t.name).isidentical(t.name.str.len())
+    assert bzs.like(t.name, '*a').isidentical(t.name.str.like('*a'))
+    assert (bzs.str_cat(bzs.str_cat(t.name, t.name, sep=' ++ '), t.name)
+            .isidentical(t.name.str.cat(t.name, sep=' ++ ')
+                               .str.cat(t.name)))
+    assert bzs.str_isalnum(t.name).isidentical(t.name.str.isalnum())
+    assert bzs.str_isalpha(t.name).isidentical(t.name.str.isalpha())
+    assert bzs.str_isdecimal(t.name).isidentical(t.name.str.isdecimal())
+    assert bzs.str_isdigit(t.name).isidentical(t.name.str.isdigit())
+    assert bzs.str_islower(t.name).isidentical(t.name.str.islower())
+    assert bzs.str_isnumeric(t.name).isidentical(t.name.str.isnumeric())
+    assert bzs.str_isspace(t.name).isidentical(t.name.str.isspace())
+    assert bzs.str_istitle(t.name).isidentical(t.name.str.istitle())
+    assert bzs.str_isupper(t.name).isidentical(t.name.str.isupper())
+
+    assert bzs.Replace(t.name, 'A', 'a').isidentical(t.name.str.replace('A', 'a'))
```

Comment:
Reason for not providing `str_replace()`, `str_capitalize()`, `str_strip()`, etc. factory functions?  If the input validation is trivial, then we can just implement the class versions with lowercased names.